sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/StatsEstimationSuite.scala

```diff
@@ -18,12 +18,41 @@
 package org.apache.spark.sql.catalyst.statsEstimation
 
 import org.apache.spark.sql.catalyst.CatalystConf
-import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeMap, AttributeReference}
-import org.apache.spark.sql.catalyst.plans.logical.{ColumnStat, LogicalPlan, Statistics}
+import org.apache.spark.sql.catalyst.expressions.{Attribute, AttributeMap, AttributeReference, Literal}
+import org.apache.spark.sql.catalyst.plans.logical._
 import org.apache.spark.sql.types.IntegerType
 
 
-class StatsConfSuite extends StatsEstimationTestBase {
+class StatsEstimationSuite extends StatsEstimationTestBase {
+  val (ar, colStat) = (attr("key"), ColumnStat(distinctCount = 10, min = Some(1), max = Some(10),
+    nullCount = 0, avgLen = 4, maxLen = 4))
+
+  val plan = StatsTestPlan(
+    outputList = Seq(ar),
+    attributeStats = AttributeMap(Seq(ar -> colStat)),
+    rowCount = 10,
+    size = Some(10 * (8 + 4)))
+
+  test("limit estimation") {
+    val localLimit = LocalLimit(Literal(2), plan)
+    val globalLimit = GlobalLimit(Literal(2), plan)
+    // LocalLimit and GlobalLimit share the same stats estimation logic.
+    val expected = Statistics(sizeInBytes = 24, rowCount = Some(2))
+    checkStats(localLimit, expected)
+    checkStats(globalLimit, expected)
+  }
+
+  test("sample estimation") {
+    val sample = Sample(0.0, 0.5, withReplacement = false, (math.random * 1000).toLong, plan)()
+    checkStats(sample, expected = Statistics(sizeInBytes = 60, rowCount = Some(5)))
+
+    // Test if Sample's child doesn't have rowCount in stats
+    val stats2 = Statistics(sizeInBytes = 120)
```

Comment:
For limit estimation test cases, we may add a test with limit number greater than a child node's row count.  This test can show if we properly select the smaller value between limit number child node's row count.